src/servers/build.rs

```diff
@@ -21,33 +21,28 @@ fn main() {
 fn fetch_dashboard_assets() {
     use std::process::{Command, Stdio};
 
-    macro_rules! p {
-        ($($tokens: tt)*) => {
-            println!("cargo:warning={}", format!($($tokens)*))
-        }
-    }
-
+    let message = "Failed to fetch dashboard assets";
+    let help = r#"
+You can manually execute './scripts/fetch-dashboard-assets.sh' to see why, 
+or it's a network error, just try again or enable/disable some proxy."#;
+    let out_dir = std::env::var("OUT_DIR").unwrap();
     let output = Command::new("./fetch-dashboard-assets.sh")
+        .arg(&out_dir)
```

Comment:
cargo provides an env to get target dir `CARGO_TARGET_DIR` https://doc.rust-lang.org/cargo/reference/environment-variables.html

Reply:
https://doc.rust-lang.org/cargo/reference/build-scripts.html?highlight=OUT_DIR#outputs-of-the-build-script

> Build scripts may save any output files or intermediate artifacts in the directory specified in the [OUT_DIR environment variable](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts). Scripts should not modify any files outside of that directory.